api-examples/transformer_utils.py

```diff
@@ -428,12 +432,29 @@ def __iter__(self):
         print(f'worker {node_worker_id} [{start_idx}:{end_idx}]')
 
         self.vectorizer.mxlen = self.nctx
+        read_file_order = np.arange(start_idx, end_idx)
+        # If we have multiple files per worker, possibly shuffle the file read order
+        if self.shuffle:
+            read_file_order = np.random.permutation(read_file_order)
 
-        for file in files[start_idx:end_idx]:
+        for file_idx in read_file_order:
+            file = files[file_idx]
             with open(file) as rf:
-                for line in rf:
-                    response = self.process_line(line)
-                    if response:
+                lines = []
+                for i, line in enumerate(rf):
+                    lines.append(line)
+                    if (i+1) % MultiFileLoader.SHUF_BUF_SZ == 0:
```

Comment:
it seems to me reading the whole file in and shuffle is good enough (for the purpose of reducing core memory usage) and simpler.